src/traces/box/plot.js

```diff
@@ -304,7 +304,7 @@ function plotBoxMean(sel, axes, trace, t) {
 
     var paths = sel.selectAll('path.mean').data((
         (trace.type === 'box' && trace.boxmean) ||
-        (trace.type === 'violin' && trace.box && trace.meanline)
+        (trace.type === 'violin' && (trace.box || {}).visible && (trace.meanline || {}).visible)
```

Comment:
can't this just be `trace.box.visible && trace.meanline.visible` given the changes below?